bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/EntsoeHandler.java

```diff
@@ -0,0 +1,272 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.entsoe.internal;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.entsoe.internal.client.Client;
+import org.openhab.binding.entsoe.internal.client.Request;
+import org.openhab.binding.entsoe.internal.exception.EntsoeConfigurationException;
+import org.openhab.binding.entsoe.internal.exception.EntsoeResponseException;
+import org.openhab.binding.entsoe.internal.exception.EntsoeResponseMapException;
+import org.openhab.core.library.dimension.Currency;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.CurrencyUnit;
+import org.openhab.core.library.unit.CurrencyUnits;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.TimeSeries;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EntsoeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jørgen Melhus - Initial contribution
+ */
+@NonNullByDefault
+public class EntsoeHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EntsoeHandler.class);
+
+    private EntsoeConfiguration config;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    private Unit<Currency> baseUnit = CurrencyUnits.BASE_CURRENCY;
+    private Unit<Currency> fromUnit = new CurrencyUnit(EntsoeBindingConstants.ENTSOE_CURRENCY, null);
+
+    private Map<ZonedDateTime, Double> responseMap;
+
+    // Use region-based ZoneId instead of CET to handle transition to CEST automatically
+    private ZonedDateTime lastDayAheadReceived = ZonedDateTime.of(LocalDateTime.MIN, ZoneId.of("Europe/Paris"));
```

Comment:
Not sure what is best/most correct, but maybe this would have the same effect?
```suggestion
    private ZonedDateTime lastDayAheadReceived = ZonedDateTime.of(LocalDateTime.MIN, ZoneId.of("CET"));
```

Also, consider storing this `ZoneId` in a constant, since it's used multiple times, and must be the same.

Reply:
I see you now store the `String` "Europe/Paris", but is there anything preventing storing the resulting `ZoneId` instead to be used directly?

Also, any comment on CET vs. Europe/Paris?

Reply:
@jlaur Did you see my comment on the CET vs. Europe/Paris, wouldn't it be easiest to keep it as Europe/Paris and instead have a comment in code explaining why that specific value is used?

Reply:
I'm pretty sure it makes no difference. In Energi Data Service binding I'm using CET, and it automatically takes DST into account.

Reply:
See:
https://github.com/openhab/openhab-addons/blob/85b165208c4ba5125bb493418362e6f5f86e84c0/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/EnergiDataServiceBindingConstants.java#L81-L85

and used for scheduling refresh time here:

https://github.com/openhab/openhab-addons/blob/85b165208c4ba5125bb493418362e6f5f86e84c0/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/handler/EnergiDataServiceHandler.java#L261-L270

Reply:
ok, changed to CET